### PR TITLE
Implement [GET] /workspaces

### LINKED ISSUE #113: Implement [GET] /workspaces/{workspace_id}
As a AzureTRE Admin, I want to retrieve a workspace request from cosmos db, so that I can populate my UI with the details

**Acceptance criteria:**

- [ ] Workspace resource retrieved from cosmos DB
- [ ] If the workspace does not exist, returns 404

### LINKED ISSUE #114: Implement [GET] /workspaces
Depends on #113 

- [ ] [GET] /workspaces retrieves all workspace resources that are not marked status=deleted

### PR DESCRIPTION
**Pull request by TessFerrandez**
# PR for issue #113 and #114 from story #62

## What is being addressed

- [GET] /workspaces implemented (to retrieve workspace resource from state store)
- [GET] /workspaces/{workspace_id} implemented (to retrieve workspace resource from state store)

## How is this addressed

- Routes implemented
- Basic Schemas and Domain models for workspaces (resources) created
- Workspace repository with get functionality implemented
- BaseRepository created that will act as the base for all repositories
- Basic DB errors, config and other related functionality that can be reused for other endpoints created
- Connection to database established on API startup
- Tests for the routes implemented
- A basic script implemented to bootstrap your cosmos DB instance with a few sample workspace resources to test locally

Closes #113 
Closes #114

### CODE REVIEW

File: management_api_app/db/repositories/workspaces.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from typing import List
+
+from azure.cosmos import CosmosClient, ContainerProxy, PartitionKey
```
Comment (msebragge) [inline management_api_app/db/repositories/workspaces.py:3]: alphabetical order: ContainerProxy, CosmosClient, PartitionKey

File: management_api_app/db/repositories/workspaces.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        workspaces = []
+        if self.container:
+            query = f'SELECT * from c WHERE c.resourceType = "{strings.RESOURCE_TYPE_WORKSPACE}" AND c.status != "{strings.RESOURCE_STATUS_DELETED}"'
+            workspaces = list(self.container.query_items(query=query, enable_cross_partition_query=True))
+        return workspaces
+
+    def get_workspace_by_workspace_id(self, workspace_id: str):
+        workspaces = []
+        if self.container:
+            query = f'SELECT * from c WHERE c.resourceType = "{strings.RESOURCE_TYPE_WORKSPACE}" AND c.status != "{strings.RESOURCE_STATUS_DELETED}" and c.id = "{workspace_id}"'
```
Comment (msebragge) [inline management_api_app/db/repositories/workspaces.py:34]: line length?

**Review by msebragge — COMMENTED** [review]
Two minor comments.

File: management_api_app/db/repositories/workspaces.py (hunk 1 above)
Comment (TessFerrandez) [reply management_api_app/db/repositories/workspaces.py:3]:
```suggestion
from azure.cosmos import ContainerProxy, CosmosClient, PartitionKey
```

File: management_api_app/db/repositories/workspaces.py (hunk 2 above)
Comment (TessFerrandez) [reply management_api_app/db/repositories/workspaces.py]: Fixed

**Review by msebragge — APPROVED** [review]

**Pull request merged**